src/Components/sharedComponents/ImageBuilderHeader.tsx

```diff
@@ -34,10 +41,32 @@ export const ImageBuilderHeader = ({
   const onBuildHandler = async () => {
     selectedBlueprint && (await buildBlueprint({ id: selectedBlueprint }));
   };
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const onSelect = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const [showDeleteModal, setShowDeleteModal] = React.useState(false);
+  const [deleteBlueprint] = useDeleteBlueprintMutation();
+  const handleDelete = async () => {
+    if (selectedBlueprint) {
+      setShowDeleteModal(false);
+      await deleteBlueprint({ id: selectedBlueprint });
+    }
+  };
+  const onDeleteClose = () => {
+    setShowDeleteModal(false);
+  };
 
   return (
     <>
       {/*@ts-ignore*/}
+      <DeleteBlueprintModal
+        onDelete={handleDelete}
+        blueprintName={''}
```

Comment:
I guess the name is missing because the component has the blueprint id. As we discussed offline, we can get the blueprint's name by creating a selector to select a blueprint's name by its id.